src/precice/impl/ParticipantImpl.hpp

```diff
@@ -327,6 +334,12 @@ class ParticipantImpl {
   /// Counts calls to advance for plotting.
   long int _numberAdvanceCalls = 0;
 
+  /// Counts the amount of samples mapped in write mappings executed in the latest advance
+  int _executedWriteMappings = 0;
+
+  /// Counts the amount of samples mapped in read mappings executed in the latest advance
+  int _executedReadMappings = 0;
```

Comment:
Is there a better way to deal with these "only for testing and debugging"-variables? It just looks a bit strange to me to have them on the same level as variables that are needed for the functionality of the code. I guess what we also do not want is that developers in the future start using these variables for implementing actual functionality. Maybe adding comment here already serves the purpose of avoiding this kind of wrong usage.
```suggestion
  /// Counts the amount of samples mapped in write mappings executed in the latest advance. Should only be used for testing.
  int _executedWriteMappings = 0;

  /// Counts the amount of samples mapped in read mappings executed in the latest advance. Should only be used for testing.
  int _executedReadMappings = 0;
```

Reply:
These were my exact thoughts too, which is why I added a debug statement, which prints this information after the coupling state.

After all, it may be interesting to see as a short summary of what has been mapped.